src/System.Private.CoreLib/shared/System/String.cs

```diff
@@ -655,6 +651,30 @@ internal static unsafe int wcslen(char* ptr)
             return count;
         }
 
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe int strlen(byte* ptr)
+        {
+            if (ptr == null)
```

Comment:
Why do we want to do this check for null?

Reply:
To mimic `lstrlenA`  https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-lstrlena
```
The function returns the length of the string, in characters. 
If lpString is NULL, the function returns 0.
```
Also there are some tests that pass null and fail with a null ref when lstrlenA is changed to strlen without it

Reply:
The handling of NULL in `lstrlenA` is Windowism. The regular C strlen does not do it. I would rather add explicit NULL checks to the few places that use `lstrlenA` today and need it.